sacred/experiment.py

```diff
@@ -378,24 +389,12 @@ def log_scalar(self, name, value, step=None):
         # The same as Run.log_scalar
         return self.current_run.log_scalar(name, value, step)
 
-    def gather_commands(self):
-        """Iterator over all commands of this experiment.
-
-        Also recursively collects all commands from ingredients.
-
-        Yields
-        ------
-        (str, function)
-            A tuple consisting of the (dotted) command-name and the
-            corresponding captured function.
-
-        """
-        for cmd_name, cmd in self.commands.items():
-            yield cmd_name, cmd
-
-        for ingred in self.ingredients:
-            for cmd_name, cmd in ingred.gather_commands():
-                yield cmd_name, cmd
+    def _gather(self, func):
+        for ingredient, _ in self.traverse_ingredients():
+            for name, item in func(ingredient):
+                if ingredient == self:
+                    name = name[len(self.path) + 1:]
```

Comment:
Not sure I understand: what is the purpose of this if clause?

Reply:
The command names of the experiment itself should not be prefixed with the experiment name. `traverse_ingredients` returns all ingredients and the gathering function returns the full names. I kept the previous behavior by removing the own path prefix from the name of the experiment.